include/json/value.h

```diff
@@ -171,6 +171,9 @@ class JSON_API Value {
     CZString(ArrayIndex index);
     CZString(const char* cstr, DuplicationPolicy allocate);
     CZString(const CZString& other);
+#ifdef JSON_HAS_RVALUE_REFERENCES
+	CZString(CZString&& other);
```

Comment:
Without `noexcept`, the move ctor often degenerates into a copy, like when reallocating a vector.

    std::vector<Json::Value> v = MakeSomeValues();
    v.reserve(v.capacity() + 1);  // realloc: all elements are deep-copied!

If a compiler supports move semantics but not `noexcept`, it needs to be special-cased. I suggest detecting and flagging noexcept support in config.h similarly to how rvalue-reference support is flagged, otherwise we're just making slow move ops where speed matters most.

There's a reasonable-looking `#if` given here:
http://stackoverflow.com/questions/18387640/how-to-deal-with-noexcept-in-visual-studio